### PR TITLE
Refactor MTE-3733 Firefox iOS tests for iOS 15 and 16

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3733)

## :bulb: Description
Some tests fail (intermittently) for older versions of iOS. Some labels for external websites have been changed.

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/TodayWidgetTests.swift [hunk 1]
```diff
@@ -323,6 +323,9 @@ class TodayWidgetTests: BaseTestCase {
     // TESTS
     // https://mozilla.testrail.io/index.php?/cases/view/2769289
     func testNewSearchWidget() throws {
+        if #unavailable(iOS 16) {
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/TodayWidgetTests.swift:326]: @mdotb-moz Sorry. I can't get the tests to pass on iOS 15. How about on your end?

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ClipBoardTests.swift [hunk 2]
```diff
@@ -41,7 +41,7 @@ class ClipBoardTests: BaseTestCase {
                     return
                 }
                 if value.hasPrefix("http") == false {
-                    value = "http://\(value)"
+                    value = "http://www.\(value)/"
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ClipBoardTests.swift:44]: iOS 15 now shows `www.` and `/` on the URL bar.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift [hunk 3]
```diff
@@ -409,6 +409,9 @@ class BookmarksTests: BaseTestCase {
         contextMenuTable.cells.otherElements[StandardImageIdentifiers.Large.privateMode].waitAndTap()
         // The webpage opens in a new private tab
         switchToTabAndValidate(nrOfTabs: "1", isPrivate: true)
+        if #unavailable(iOS 16) {
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift:412]: iOS 15 only: The keyboard is raised when the private tab is opened.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ScreenGraphTest.swift [hunk 4]
```diff
@@ -168,15 +168,15 @@ private func createTestGraph(for test: XCTestCase, with app: XCUIApplication) ->
 
     map.addScreenState(ToolsBrowserTabMenu) { screenState in
         screenState.tap(
-            app.otherElements.images[StandardImageIdentifiers.Large.nightMode],
+            app.tables.cells[AccessibilityIdentifiers.MainMenu.nightMode],
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ScreenGraphTest.swift:171]: iOS 15 and 16 don't have identifier for the images, so we use the cells.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ToolbarMenuTests.swift [hunk 5]
```diff
@@ -48,11 +48,11 @@ class ToolbarMenuTests: BaseTestCase {
         mozWaitForElementToExist(hamburgerMenu)
         mozWaitForElementToExist(app.textFields[AccessibilityIdentifiers.Browser.AddressToolbar.searchTextField])
         mozWaitForElementToExist(tabsButton)
+        mozWaitForElementToExist(firstPocketCell)
         XCTAssertTrue(
             hamburgerMenu.isRightOf(rightElement: tabsButton),
             "Menu button is not on the right side of tabs button"
         )
-        mozWaitForElementToExist(firstPocketCell)
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ToolbarMenuTests.swift:55]: When the device is getting rotated, the element exist but may need more time to get to the positions.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ActivityStreamTest.swift [hunk 6]
```diff
@@ -12,7 +12,7 @@ let newTopSite = [
     "bookmarkLabel": "Mozilla - Internet for people, not profit (US)"
 ]
 let newTopSiteiOS15 = [
-    "bookmarkLabel": "Internet for people, not profit — Mozilla (US)"
+    "bookmarkLabel": "Mozilla — Internet for people, not profit"
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ActivityStreamTest.swift:15]: 🤷🏼‍♀️ Some labels for websites have been updated.

**Review by isabelrios — APPROVED** [review]

**Pull request merged**